main.tf

```
@@ -44,360 +61,45 @@ module "snowflake_stage" {
   snowflake_iam_user  = each.value.snowflake_iam_user
   storage_integration = each.value.storage_integration
   url                 = each.value.url
+  roles               = each.value.roles
 
-  create_default_roles = coalesce(each.value.create_default_roles, var.create_default_roles)
-  roles                = each.value.roles
+  create_default_roles = each.value.create_default_roles
 }
 
 module "snowflake_default_role" {
   for_each = local.default_roles
 
-  source  = "getindata/role/snowflake"
-  version = "1.3.0"
+  source  = "getindata/database-role/snowflake"
+  version = "1.1.1"
   context = module.this.context
-  enabled = local.create_default_roles && each.value.enabled
+  enabled = local.create_default_roles && lookup(each.value, "enabled", true)
 
-  name       = each.key
-  attributes = [local.database, local.schema]
+  database_name = one(snowflake_schema.this[*].database)
+  name          = each.key
+  attributes    = [local.database, local.schema]
```

We don't need DB name in DB role attributes

```suggestion
  attributes    = [local.schema]
```